iron-list.html

```diff
@@ -1244,6 +1244,15 @@
         var item = this.items && this.items[vidx];
         if (item != null) {
           inst[this.as] = item;
+
+          if(typeof item === 'object' && !Array.isArray(item)) {
+            for(var prop in item) {
+              if(item.hasOwnProperty(prop)) {
+                inst.set([this.as, prop], item[prop]);
```

Comment:
Sorry for the long time to respond. The main use case I see where that might affect perf is some sort of filtering, e.g. immediate results per every keystroke in a search box. I'll ask @danbeam about this.